src/data_gradients/config/data/data_config.py

```diff
@@ -37,18 +37,53 @@ class DataConfig(ABC):
 
     def __post_init__(self):
         self.cache_dir = self.cache_dir if self.cache_dir is not None else self.DEFAULT_CACHE_DIR
-
-        # Once the object is initialized, we check if the cache is activated or not.
         if self.cache_filename is not None:
-            cache_path = os.path.join(self.cache_dir, self.cache_filename)
-            logger.info(
-                f"Cache activated for `{self.__class__.__name__}`. This will be used to set attributes that you did not set manually. "
-                f'Caching to "{cache_path}"'
-            )
-            self._fill_missing_params_with_cache(cache_path)
+            # Once the object is initialized, we check if the cache is activated or not.
+            self.update_from_cache_file()
         else:
             logger.info(f"Cache deactivated for `{self.__class__.__name__}`.")
 
+    @property
+    def is_cache_file_used(self):
+        return self.cache_filename is not None
+
+    @property
+    def cache_path(self):
+        if not self.cache_filename:
+            raise ValueError(f"Cannot load/save cache from `{self.__class__.__name__}`. Please set `cache_filename=...`")
+        return os.path.join(self.cache_dir, self.cache_filename)
+
+    def update_from_cache_file(self):
+        """Update the values that are not set yet, using the cache file."""
+        if os.path.isfile(self.cache_path):
+            print(
+                f"Using cache to update `{self.__class__.__name__}` from:\n"
+                f"    - cache_dir:      {self.cache_dir}\n"
+                f"    - cache_filename: {self.cache_filename}"
+            )
+            self._fill_missing_params_with_cache(self.cache_path)
+        else:
+            logger.warning(
+                f"Expected cache file at {self.cache_path} but none was found. Ensure the correct path is set. "
+                f"You can set `{self.__class__.__name__}(cache_filename=..., cache_dir=...)`."
+            )
+
+    def dump_cache_file(self):
```

Comment:
Tend to agree that it is not good to print whenever we write to config file.
But we cannot have 2 function for each functionality that may need a print statement.
Instead I propose to have these function return the "Summary" (`str`) of what happened. This can then be used to print or simply ignored.
See this commit - the final stdout is the same as before. https://github.com/Deci-AI/data-gradients/pull/186/commits/d426f0b026734cfc2c474b4a655990ad69923c0e